users/serializers.py

```diff
@@ -0,0 +1,35 @@
+from rest_framework import serializers
+from rest_framework.compat import authenticate
+from .models import User
+
+class UserAuthSerializer(serializers.Serializer):
+    email = serializers.EmailField(label='Email Address')
+    password = serializers.CharField(
+        label='Password',
+        style={'input_type': 'password'},
+        trim_whitespace=False,
+    )
+
+    def validate(self, data):
+        email = data.get('email')
+        password = data.get('password')
+
+        if email and password:
+            user_qs = User.objects.filter(email=email)
+
+            if user_qs.count()==1:
```

Comment:
Notes:
1. queryset returns a list of objects. you can directly check if it is empty or not by just `if user_qs:`
2. negation technique is best for this kind of situations.
```
users = User.objects.filter(email=email)
user = users.first()

if not users or not user.check_password(password) or not user.is_active:
    raise serializers.ValidationError("Email/Password is incorrect. Please try again.")

return data
```